dotCMS/src/main/java/com/dotmarketing/servlets/BinaryExporterServlet.java

```diff
@@ -192,123 +194,123 @@ public void doGet(HttpServletRequest req, HttpServletResponse resp) throws Servl
-						StringBuilder query = new StringBuilder();
-						query.append("+(languageId:").append(defaultLang).append(" languageId:").append(lang).append(") ");
-						query.append("+identifier:").append(assetIdentifier).append(" +deleted:false ");
-						if ( live ) {
-							query.append("+live:true ");
-						} else {
-							query.append("+working:true ");
-						}
-
-						List<Contentlet> foundContentlets = contentletAPI.search(query.toString(), 2, -1, null, user, respectFrontendRoles);
-						if ( foundContentlets != null && !foundContentlets.isEmpty() ) {
-							//Prefer the contentlet with the session language
-							content = foundContentlets.get(0);
-							if ( content.getLanguageId() != lang && foundContentlets.size() == 2 ) {
-								content = foundContentlets.get(1);
-							}
-						} else {
-							Logger.error(this, "Content with Identifier [" + assetIdentifier + "] not found.");
-							resp.sendError(404);
-							return;
-						}
-
-					}
-					else {
-						/*
+			Contentlet content = null;
+			
+			if(byInode) {
+			    content = (isTempBinaryImage) ? contentAPI.find(assetInode, APILocator.systemUser(), respectFrontendRoles)
+			            :contentAPI.find(assetInode, user, respectFrontendRoles);
+			    assetIdentifier = content.getIdentifier();
+			} else {
+			    isLiveMode = userWebAPI.isLoggedToFrontend(req);
+			    boolean PREVIEW_MODE = false;
+			    boolean EDIT_MODE = false;
+
+			    if(session != null) {
+			        PREVIEW_MODE = (session.getAttribute(com.dotmarketing.util.WebKeys.PREVIEW_MODE_SESSION) != null);
+			        try {
+			            EDIT_MODE = (session.getAttribute(com.dotmarketing.util.WebKeys.EDIT_MODE_SESSION) != null);
+			        } catch (Exception e) {
+			            Logger.error(this, "Error: Unable to determine if there's a logged user.", e);
+			        }
+			    }
+			    //GIT-4506
+			    if(WebAPILocator.getUserWebAPI().isLoggedToBackend(req)){
+			        isLiveMode = (!EDIT_MODE && !PREVIEW_MODE);
+			    }
+
+			    if (req.getSession(false) != null && req.getSession().getAttribute("tm_date")!=null) {
+			        isLiveMode = Boolean.TRUE;
+			        Identifier ident=APILocator.getIdentifierAPI().find(assetIdentifier);
+			        if(UtilMethods.isSet(ident.getSysPublishDate()) || UtilMethods.isSet(ident.getSysExpireDate())) {
+			            Date fdate = new Date(Long.parseLong((String)req.getSession().getAttribute("tm_date")));
+			            isLiveMode = ifIdentifierIsUnpublishedOrExpired(ident.getSysPublishDate(),fdate);
+			            isContentExpired = ifIdentifierIsUnpublishedOrExpired(ident.getSysExpireDate(),fdate);
+			        }
+                    if(isContentExpired){
+                        Logger.debug(this, "Id " + assetIdentifier + " belongs to an expired content.");
+                        resp.sendError(HttpServletResponse.SC_NOT_FOUND);

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Add a "try/catch" block for "sendError". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1989)